contract/src/adjudicators/Counter.sol

```diff
@@ -24,99 +24,86 @@ contract Counter is IAdjudicator {
-     * @param chan The channel configuration
-     * @param candidate The proposed counter state
-     * @param proofs Array containing the previous state signed by the previous participant
-     * @return decision The status of the channel after adjudication
+     * @notice Validates that the counter state transition is valid with strict turn‐taking.
+     * @param chan The channel configuration.
+     * @param candidate The proposed new state.
+     * @param proofs Array containing the previous state.
+     * @return valid True if the state transition is valid, false otherwise.
      */
     function adjudicate(Channel calldata chan, State calldata candidate, State[] calldata proofs)
         external
-        pure
+        view
         override
-        returns (Status decision)
+        returns (bool valid)
     {
-        // Check if we have at least one signature
-        if (candidate.sigs.length == 0) return Status.INVALID;
+        // Ensure the candidate state is signed by both participants.
+        if (candidate.sigs.length != 2) {
+            return false;
+        }
 
-        // Get the state hash for signature verification
+        // Compute the state hash for signature verification.
         bytes32 stateHash = Utils.getStateHash(chan, candidate);
 
-        // Decode the counter from candidate state.data
-        CounterData memory candidateCounterData = abi.decode(candidate.data, (CounterData));
-
-        // INITIAL STATE ACTIVATION
-        if (candidateCounterData.counter == 0) {
-            // First signature must be from HOST who sets initial counter
-            if (!Utils.verifySignature(stateHash, candidate.sigs[0], chan.participants[HOST])) {
-                return Status.VOID;
-            }
-
-            // If only Host has signed, channel is PARTIAL
-            if (candidate.sigs.length < 2) {
-                return Status.PARTIAL;
-            }
+        // Decode the candidate state data.
+        CounterApp memory candidateState = abi.decode(candidate.data, (CounterApp));
 
-            // Both signatures provided, verify Guest's signature
-            if (!Utils.verifySignature(stateHash, candidate.sigs[1], chan.participants[GUEST])) {
-                return Status.VOID;
-            }
-
-            // Channel becomes ACTIVE only if counter > 0
-            if (candidateCounterData.counter > 0) {
-                return Status.ACTIVE;
-            } else {
-                return Status.PARTIAL;
+        // INITIAL STATE: version 0 requires both signatures.
+        if (candidateState.version == 0) {
+            if (
+                !Utils.verifySignature(stateHash, candidate.sigs[0], chan.participants[HOST])
+                    || !Utils.verifySignature(stateHash, candidate.sigs[1], chan.participants[GUEST])
+            ) {
+                return false;
             }
+            // Valid initial state.
+            return true;
```

Comment:
This can be simplified to `return Utils.verifySignature(...) && Utils.verifySignature(...);`

Reply:
c3a15f89e51784a39abaa078c81481a3aa8a1b7f